bbq/compiler/compiler.go

```diff
@@ -2036,14 +2036,12 @@ func isDynamicMethodInvocation(accessedType sema.Type) bool {
 func (c *Compiler[_, _]) compileArguments(arguments ast.Arguments, invocationTypes sema.InvocationExpressionTypes) {
 	for index, argument := range arguments {
 		c.compileExpression(argument.Expression)
-		c.emitTransfer(invocationTypes.ParameterTypes[index])
+		if invocationTypes.ParameterTypes[index] == nil {
+			c.emitTransfer()
+		} else {
+			c.emitTransferAndConvert(invocationTypes.ParameterTypes[index])
+		}
```

Comment:
Test and use the same variable:

```suggestion
		parameterType := invocationTypes.ParameterTypes[index]
		if parameterType == nil {
			c.emitTransfer()
		} else {
			c.emitTransferAndConvert(parameterType)
		}
```